os/include/tinyara/pm/pm.h

```diff
@@ -208,60 +200,49 @@ struct pm_callback_s {
 	char name[MAX_PM_CALLBACK_NAME];	/* Name of driver which register callback */
 
 	/**************************************************************************
-	 * Name: prepare
+	 * Name: sleep
 	 *
 	 * Description:
-	 *   Request the driver to prepare for a new power state. This is a
-	 *   warning that the system is about to enter into a new power state.  The
+	 *   Put the driver into low power mode (or turn OFF). This is a
+	 *   warning that the system is about to enter into PM_SLEEP state. The
 	 *   driver should begin whatever operations that may be required to enter
-	 *   power state.  The driver may abort the state change mode by returning
+	 *   PM_SLEEP state. The driver may abort the state change mode by returning
 	 *   a non-zero value from the callback function
 	 *
 	 * Input Parameters:
 	 *   cb      - Returned to the driver.  The driver version of the callback
 	 *             structure may include additional, driver-specific state
 	 *             data at the end of the structure.
-	 *   domain  - Identifies the activity domain of the state change
-	 *   pmstate - Identifies the new PM state
 	 *
 	 * Returned Value:
 	 *   0 (OK) means the event was successfully processed and that the driver
-	 *   is prepared for the PM state change.  Non-zero means that the driver
+	 *   is prepared for the PM_SLEEP change.  Non-zero means that the driver
 	 *   is not prepared to perform the tasks needed achieve this power setting
-	 *   and will cause the state change to be aborted.  NOTE:  The prepare
-	 *   method will also be recalled when reverting from lower back to higher
-	 *   power consumption modes (say because another driver refused a lower
-	 *   power state change).  Drivers are not permitted to return non-zero
-	 *   values when reverting back to higher power consumption modes!
+	 *   and will cause the state change to be aborted.
 	 *
 	 **************************************************************************/
 
-	int (*prepare)(FAR struct pm_callback_s *cb, enum pm_state_e pmstate);
+	int (*sleep)(FAR struct pm_callback_s *cb);
```

Comment:
do we need prepare?

Reply:
`prepare` is not needed in new design because we have `wake` callback now. If before sleep, a driver failed to change its state according to the new PM state, it will raise `ERROR` then PM will wake all the slept driver immediately. The drivers must support `wake` operation. The `prepare` is only needed if we want to separate prepare functionality from `sleep` callback. Considering current `NDP` and `Audio` driver code this separation is not needed.